index.js

```
@@ -88,7 +88,58 @@ let projects = {};
+      // changes then "A" does not to be set up).
+      const projectSetupGroups = [];
+      Object.keys(config.projects).forEach((projectName) => {
+        const project = config.projects[projectName];
+        if (project.dependsOnProjects) {
+          project.dependsOnProjects.forEach((dependentProjectName) => {
+            if (!config.projects[dependentProjectName]) {
+              throw new Error(`Couldn't find project "${dependentProjectName}" that "${projectName}" depends on`);
+            }
+            if (config.projects[dependentProjectName].dependsOnProjects) {
+              throw new Error(`"${projectName}" depends on "${dependentProjectName}", which itself depends on other projects, cannot do that`);
+            }
+          });
+          projectSetupGroups.push(project.dependsOnProjects.concat([projectName]));
+        }
+      });
+      log(`All project groups for setup: ${projectSetupGroups.map((group) => group.join('+')).join(', ')}...`);
+
+      // If a project is used, make sure its whole group is used.
+      Object.keys(projects).forEach((projectName) => {
+        projectSetupGroups.forEach((group) => {
+          if (group.includes(projectName)) {
+            log(`Using project "${projectName}", so for setup using all of ${group.join('+')}...`);
+            group.forEach((groupProjectName) => {
+              projects[groupProjectName] = config.projects[groupProjectName];
+            });
+          }
+        });
+      });
+
+      log(`Using projects (just for setup): ${Object.keys(projects).join(', ')}...`);
+    } else {
+      // If a project is used, make sure projects that depend on it are used.
+      Object.keys(config.projects).forEach((projectName) => {
+        const project = config.projects[projectName];
+        if (project.dependsOnProjects) {
+          project.dependsOnProjects.forEach((dependentProjectName) => {
+            if (projects[dependentProjectName]) {
```

when would this be false? Silently passing on this is dangerous, probably..